python/cuml/neighbors/nearest_neighbors.pyx

```diff
@@ -268,6 +272,9 @@ cdef class NearestNeighbors:
         if self.k != NULL:
             del self.k
 
+        if self.X_m != NULL:
+            del self.X_m
+
```

Comment:
Do you also need to set self.X_m = NULL or does cython do that automagically?

Reply:
I initialize X_m and k to NULL explicitly because I'm testing for NULL here when the fit() function has been called more than once. I personally prefer the explicit initialization for readability but I should caution that I'm far from a cython expert.

Reply:
Cython docs state that by the time `__cinit__()` is called, the values have already been initialized so I suppose the NULL is redundant.